test/common/creation.go

```diff
@@ -0,0 +1,172 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"github.com/knative/eventing/pkg/apis/eventing/v1alpha1"
+	"github.com/knative/eventing/test/base"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+)
+
+var coreAPIGroup = corev1.SchemeGroupVersion.Group
+var coreAPIVersion = corev1.SchemeGroupVersion.Version
+var rbacAPIGroup = rbacv1.SchemeGroupVersion.Group
+var rbacAPIVersion = rbacv1.SchemeGroupVersion.Version
+
+// CreateChannel will create a Channel Resource in Eventing.
+func (client *Client) CreateChannel(name, provisonerName string) error {
+	namespace := client.Namespace
+	channel := base.Channel(name, provisonerName)
+
+	channels := client.Eventing.EventingV1alpha1().Channels(namespace)
+	// update channel with the new reference
+	channel, err := channels.Create(channel)
+	if err != nil {
+		return err
+	}
+	client.Cleaner.AddObj(channel)
+	return nil
```

Comment:
This is okay, but you could also use argument unpacking to collapse lines 38-43 to:

```
return client.Cleaner.AddObj(channels.Create(channel))
```

This would require that AddObj takes a second `error` argument and returns an `error` (if the error provided is non-nil). Overall, this pattern occurs 4-6 times in the code, so you'd save about 20 lines. I don't feel strongly about this one.

Reply:
I would prefer not to change since it looks more natural to make the `Add` and `AddObj` function similar :-)